orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java

```diff
@@ -691,132 +678,142 @@ void streamChatCompletionDeltas() throws IOException {
+    assertThat(orchestrationResult.getSystemFingerprint()).isEqualTo("fp_f3927aa00d");
     assertThat(orchestrationResult.getChoices()).hasSize(1);
     assertThat(orchestrationResult.getChoices().get(0).getMessage().getContent())
         .isEqualTo(
-            "Of course! Just let me put on my human glasses... Oh wait, I left them in the matrix");
+            "Well, this image features the logo of SAP, a multinational software corporation known for its enterprise resource planning (ERP) software. The logo is prominently displayed in white against a gradient blue background.");
     assertThat(orchestrationResult.getChoices().get(0).getFinishReason()).isEqualTo("stop");
     assertThat(orchestrationResult.getChoices().get(0).getMessage().getRole())
         .isEqualTo("assistant");
     assertThat(orchestrationResult.getChoices().get(0).getIndex()).isZero();
-    assertThat(orchestrationResult.getUsage().getCompletionTokens()).isEqualTo(31);
-    assertThat(orchestrationResult.getUsage().getPromptTokens()).isEqualTo(928);
-    assertThat(orchestrationResult.getUsage().getTotalTokens()).isEqualTo(959);
+    assertThat(orchestrationResult.getUsage().getCompletionTokens()).isEqualTo(38);
+    assertThat(orchestrationResult.getUsage().getPromptTokens()).isEqualTo(243);
+    assertThat(orchestrationResult.getUsage().getTotalTokens()).isEqualTo(281);
 
-    try (var requestInputStream = fileLoader.apply("multiChatMessageRequest.json")) {
+    try (var requestInputStream = fileLoader.apply("multiMessageRequest.json")) {
       final String requestBody = new String(requestInputStream.readAllBytes());
       verify(
           postRequestedFor(urlPathEqualTo("/completion"))
               .withRequestBody(equalToJson(requestBody)));
     }
   }
 
-  @SneakyThrows
   @Test
-  void testOrchestrationChatResponseWithMultiChatMessage() {
-    var module = new SimpleModule();
-    module.setMixInAnnotation(LLMModuleResult.class, JacksonMixins.NoneTypeInfoMixin.class);
-    module.addDeserializer(
-        LLMModuleResult.class,
-        PolymorphicFallbackDeserializer.fromJsonSubTypes(LLMModuleResult.class));
-    module.setMixInAnnotation(ChatMessage.class, JacksonMixins.NoneTypeInfoMixin.class);
-    module.addDeserializer(
-        ChatMessage.class, PolymorphicFallbackDeserializer.fromJsonSubTypes(ChatMessage.class));
-
-    var orchestrationChatResponse =
-        new OrchestrationChatResponse(
-            new ObjectMapper()
-                .registerModule(module)
-                .readValue(
-                    new String(
-                        fileLoader.apply("__files/multiChatMessageResponse.json").readAllBytes()),
-                    CompletionPostResponse.class));
-
-    assertThatThrownBy(orchestrationChatResponse::getAllMessages)
-        .isInstanceOf(UnsupportedOperationException.class)
-        .hasMessage("Messages of MultiChatMessage type not supported by convenience API");
+  void testMessageConstruction() {
```

Comment:
good to cover this with a test 👍🏻 

If you like, you could also move this to a separate test file, since it doesn't need the client and this test is already long 😄